src/main/java/com/minecolonies/coremod/research/GlobalResearchTree.java

```diff
@@ -94,4 +94,18 @@ public void loadCost()
     {
         researchTree.values().forEach(b -> b.values().forEach(IGlobalResearch::loadCostFromConfig));
     }
+
+    @Override
+    public String getEffectIdForResearch(String id)
```

Comment:
@NotNull final

Reply:
It should be nullable, it will just return null in that case. But I'll add the final.

Reply:
thenThen @Nullable

Reply:
Shouldn't that be on the interface?

Reply:
I think this should be resolved.

Reply:
in the interface it says @NotNull though =D

Reply:
Yeah, I decided it was safe to do that.